src/main/java/com/google/jenkins/plugins/k8sengine/CredentialsUtil.java

```diff
@@ -0,0 +1,120 @@
+import com.google.jenkins.plugins.k8sengine.client.Messages;
+import hudson.AbortException;
+import hudson.model.ItemGroup;
+import hudson.security.ACL;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import jenkins.model.Jenkins;
+
+/** Provides a library of utility functions for credentials-related work. */
+public class CredentialsUtil {
+  /**
+   * Get the Google Robot Credentials for the given credentialsId.
+   *
+   * @param itemGroup A handle to the Jenkins instance. Must be non-null.
+   * @param domainRequirements A list of domain requirements. Must be non-null.
+   * @param credentialsId The ID of the GoogleRobotCredentials to be retrieved from Jenkins and
+   *     utilized for authorization. Must be non-empty or non-null and exist in credentials store.
+   * @return Google Robot Credential for the given credentialsId.
+   * @throws AbortException If there was an issue retrieving the Google Robot Credentials.
+   */
+  public static GoogleRobotCredentials getRobotCredentials(
+      ItemGroup itemGroup,
+      ImmutableList<DomainRequirement> domainRequirements,
+      String credentialsId)
+      throws AbortException {
+    Preconditions.checkNotNull(itemGroup);
+    Preconditions.checkNotNull(domainRequirements);
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(credentialsId));
+
+    GoogleRobotCredentials robotCreds =
+        CredentialsMatchers.firstOrNull(
+            CredentialsProvider.lookupCredentials(
+                GoogleRobotCredentials.class, itemGroup, ACL.SYSTEM, domainRequirements),
+            CredentialsMatchers.withId(credentialsId));
+
+    if (robotCreds == null) {
+      throw new AbortException(Messages.ClientFactory_FailedToRetrieveCredentials(credentialsId));
+    }
+
+    return robotCreds;
+  }
+
+  /**
+   * Get the Credential from the Google robot credentials for GKE access.
+   *
+   * @param robotCreds Google Robot Credential for desired service account.
+   * @return Google Credential for the service account.
+   * @throws AbortException if there was an error initializing HTTP transport.
+   */
+  public static Credential getGoogleCredential(GoogleRobotCredentials robotCreds)
+      throws AbortException {
+    Credential credential;
+    try {
+      credential = robotCreds.getGoogleCredential(new ContainerScopeRequirement());
+    } catch (GeneralSecurityException gse) {
+      throw new AbortException(
+          Messages.ClientFactory_FailedToInitializeHTTPTransport(gse.getMessage()));
+    }
+
+    return credential;
+  }
+
+  /**
+   * Wrapper to get access token for service account with this credentialsId.
+   *
+   * @param credentialsId The service account credential's id. Must be non-null.
+   * @return Access token from OAuth to allow kubectl to interact with the cluster.
+   * @throws java.io.IOException If an error occurred fetching the access token.
+   */
+  static String getAccessToken(String credentialsId) throws IOException {
```

Comment:
In `doCheckCredentialsId` from [`KubernetesEngineBuilder`](https://github.com/jenkinsci/google-kubernetes-engine-plugin/blob/48a2c458faadecec498380cfd1c27bdf2445a393/src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java#L332) we currently call `getClientFactory` for verifying the credentials. I think this should be replaced with this version of `getAccessToken`, and `AbortException` in the catch should be replaced with `IOException`.

Reply:
Done.